building failed on MacOS High Sierra 10.13.2

When running `make`, after a smooth `./configure`. Throws two errors:

```
/Library/Developer/CommandLineTools/usr/include/c++/v1/__functional_base:61:21: error: invalid operands to binary expression
      ('const stellar::PublicKey' and 'const stellar::PublicKey')
        {return __x < __y;}
```
and

```
/Library/Developer/CommandLineTools/usr/include/c++/v1/__tree:1988:22: error: no matching function for call to object of type 'value_compare'
      (aka 'std::__1::__map_value_compare<stellar::PublicKey, std::__1::__value_type<stellar::PublicKey, std::__1::shared_ptr<stellar::Peer> >,
      std::__1::less<stellar::PublicKey>, true>')
            else if (value_comp()(__nd->__value_, __v))
```

Resulting in:

```
2 warnings and 2 errors generated.
make[3]: *** [overlay/Floodgate.o] Error 1
make[2]: *** [all] Error 2
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

--- a/src/overlay/Floodgate.cpp
+++ b/src/overlay/Floodgate.cpp
@@ -16,6 +16,8 @@
 namespace stellar
 {
 
+using xdr::operator<;
+
 Floodgate::FloodRecord::FloodRecord(StellarMessage const& msg, uint32_t ledger,
                                     Peer::pointer peer)
     : mLedgerSeq(ledger), mMessage(msg)

